src/signature.rs

```diff
@@ -0,0 +1,93 @@
+//! Device electronic signature
+//!
+//! (stored in system flash memory)
+
+/// This is the test voltage, in millivolts of the calibration done at
+/// the factory
+pub const VDDA_CALIB: u32 = 3300;
+
+macro_rules! define_ptr_type {
```

Comment:
Aren't these locations something that should go into the PAC crate? The interface is "register-like" and other crates like `stm32ral` would benefit just as well.

Reply:
There's certainly an argument for that, it's just that ST don't ever put them in SVDs (since they're not a 'peripheral'). So my slight preference is to leave them out of the PAC crate and implement like this.

This solution was inspired by the [f4](https://github.com/stm32-rs/stm32f4xx-hal/pull/89) crate, but there's more direct ways of doing it (like [f0](https://github.com/stm32-rs/stm32f0xx-hal/blob/master/src/adc.rs#L40)).

Reply:
Ok. If there is precedent. Don't know whether the f0 style being more compact or the f4 style of being more functional is better.

Reply:
Also relevant https://github.com/stm32-rs/stm32-rs/issues/245